### PR TITLE
Misc package cleanup, mostly lints

### PR DESCRIPTION
**Pull request by kevmoo**
Centralized HTTP JSON download logic

### CODE REVIEW

File: tool/update_shared.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  try {
+    final request = await client.getUrl(Uri.parse(uri));
+    final response = await request.close();
+
+    return response
+        .transform(utf8.decoder)
+        .transform(const JsonDecoder())
+        .single;
+  } finally {
+    client.close();
```
Comment (kevmoo) [inline tool/update_shared.dart:19]: Seeing the same bug three times put me over the edge! Also it's nice to use our stream API

File: benchmark/benchmark.dart [hunk 2]
```diff
@@ -38,7 +38,8 @@ void main() {
     // the VM doesn't optimize "dead" code away.
     if (result != expected) {
       print('Incorrect output:\n$result');
-      exit(1);
+      exitCode = 1;
```
Comment (kevmoo) [inline benchmark/benchmark.dart:41]: `exit(...)` makes debugging impossible. Cleaned these up, too!

File: test/util.dart [hunk 3]
```diff
@@ -24,19 +23,13 @@ Future<void> testDirectory(String name, {ExtensionSet? extensionSet}) async {
   }
 }
 
-Future<String> get markdownPackageRoot async {
```
Comment (kevmoo) [inline test/util.dart:27]: We can just assume tests are run from the root of the package. This vestigial logic is very much no longer needed.

**Review by devoncarew — APPROVED** [review]

File: test/util.dart (hunk 3 above)
Comment (devoncarew) [reply test/util.dart:27]: This may be causing test failures for the roll to google3.

We should revert this (or filter out related tests in google3).

Comment (kevmoo) [reply test/util.dart:27]: Oh. Ugh.

**Pull request merged**